test/ruby/test_zjit.rb

```diff
@@ -843,6 +843,16 @@ def test_instruction_order
     end
   end
 
+  def test_set
+    assert_compiles '3', %q{
+      def test_set
+        s = Set.new([1, 2, 3])
```

Comment:
This isn't testing anything in your PR, really, at least until you add cruby_methods stuff. And even then you will probably want to do the constants trick because the JIT can't (yet) see through Class#new